src/views/tutorial-view/server.ts

```diff
@@ -34,14 +37,33 @@ export async function getTutorialPageProps(
    * from the params. So we can assume `slug` index 1 is always the tutorial name
    * */
   const [collectionFilename, tutorialFilename] = slug
-  const tutorialDbSlug = `${product.slug}/${tutorialFilename}`
-  const baseTutorialData = await getTutorial(tutorialDbSlug)
+  const collectionDbSlug = `${product.slug}/${collectionFilename}`
+  const currentCollection = await getCollection(collectionDbSlug)
+  /**
+   * We need to get the database slug for this tutorial, which may belong in a different
+   * product directory in the filesystem. For example a tutorial with slug : `consul/get-started`
+   * may be reference in a vault collection. So we can't assume this current
+   * product context is valid for the db slug path
+   *
+   * @TODO - FOR PROD We will add a check in the content sync that
+   * ensures no two files in a collection have the same filename
+   */
+  const currentTutorial = currentCollection.tutorials.find((t) =>
+    t.slug.endsWith(tutorialFilename)
+  )
+
+  if (!currentTutorial) {
```

Comment:
Other error handling for requesting API data is handled in the client. This is a simple protection for this filter, could use some more thought on how best to catch issues. I assume with use we'll see errors and improve the handling as we go.